nifi-nar-bundles/nifi-ocr-bundle/nifi-ocr-processors/src/main/java/org/apache/nifi/processors/ocr/TesseractOCRProcessor.java

```diff
@@ -0,0 +1,361 @@
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TESSERACT_PAGE_SEG_MODE = new PropertyDescriptor
+            .Builder().name("Tesseract Page Segmentation Mode")
+            .description("Set Tesseract to only run a subset of layout analysis and assume a certain form of image.")
+            .required(true)
+            .defaultValue(PAGE_SEGMENTATION_MODES.get(3).getValue())
+            .allowableValues(PAGE_SEGMENTATION_MODES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TESSERACT_CONFIGS = new PropertyDescriptor
+            .Builder().name("Tesseract configuration values")
+            .description("Comma separated list of key=value pairs that will be used to configure the Tesseract instance." +
+                    " If a Tesseract configuration file is specified that will take precedence over these configurations. Values" +
+                    " placed into this property will not be validated so take care to pass only valid Tesseract configuration values." +
+                    " EX: textord_min_linesize=3.25,tessedit_write_images=true")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("successfully completed OCR on image")
+            .build();
+
+    public static final Relationship REL_UNSUPPORTED_IMAGE_FORMAT = new Relationship.Builder()
+            .name("unsupported image format")
+            .description("The image format in the FlowFile content is not supported by Tesseract")
+            .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The original image that OCR was performed on")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Failed to attempt OCR on input image")
+            .build();
+
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(TESS_DATA_PATH);
+        descriptors.add(TESSERACT_LANGUAGE);
+        descriptors.add(TESSERACT_PAGE_SEG_MODE);
+        descriptors.add(TESSERACT_CONFIGS);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_UNSUPPORTED_IMAGE_FORMAT);
+        relationships.add(REL_ORIGINAL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+    }
```

Comment:
So, there is an outstanding issue with NiFi that makes the above pattern a bit problematic (although I agree it is a common pattern in existing components). You can read more here https://issues.apache.org/jira/browse/NIFI-1318. So I'd suggest refactoring and moving it into static block. You can use AMQP, new JMS, Kafka etc as examples of what I mean.